### ISSUE TITLE
aria-controls for adjacent/contiguous DOM nodes

### ISSUE THREAD

**Issue by electronicwoft** [w3c/aria-practices#1009]
the accordian example uses aria-controls on the button, but the accordian content immediately follows it. focus can be moved into the accordian content easily and immediately without needing to move it programmatically. is there any need for aria-controls in which case?

**Comment by Decrepidos** [comment]
There are two reasons I can think of where aria-controls is useful on the accordion. Firstly it may be visually obvious that the content follows the button, however for a blind user this may not be the case. Aria-controls helps with this association. In addition some screen readers like JAWS have keyboard shortcuts to help users move to the controlled area. [JAWS key + Alt + M]

**Comment by electronicwoft** [comment]
Perhaps this is a question of how this attribute is supported? As an everyday  screen reader user, I expect that, once something is expanded, then content will immediately follow, and, rather than providing a keystroke for moving focus, maybe this should be bound to a simple keystroke such as the down arrow? This is the behaviour, for example, for an HTML <select> element. Putting the way this attribute is supported aside, my point is that aria-controls is for non-contiguous items whereas, in the example, the header for the accordian and its contents are contiguous/immediate siblings in the DOM. Would, then, aria-owns be a more appropriate attribute (conceptually and visually the content is a child of its parent header)?

**Comment by Decrepidos** [comment]
@accdc any input to clarify this one :-)

**Comment by accdc** [comment]
Hi,
This is actually more of a spec compliance issue than a usability one.

The ARIA spec states that, when aria-expanded is used to control any type of dynamic content, aria-controls should be used to referenced the controlled container or widget. This is meant to provide additional functionality for ATs that wish to do something with that information as well, but how this is used varies by AT, and may or may not be practically useful in many cases. This is why the attribute is present here though.

The aria-owns attribute, in contrast, must never be used like this simply to associate adjacent widgets, because doing so will forcibly change the accessibility tree and can cause critical accessibility issues to occur, such as the stuffing of interactive elements into other active elements; making them impossible to use accessibly.

**Comment by electronicwoft** [comment]
Quoteing ARIA 1.1 regarding the aria-expanded state: "Indicates whether the element, or another grouping element it controls, is currently expanded or collapsed". It is arguable that an accordian is a unitary element for which aria-expanded indicates the state of the accordian and not the expanded state of something that the accordian controls. In any case, what happens if the expandible part of an accordian is implemented as a child of its header?

**Comment by accdc** [comment]
"In any case, what happens if the expandible part of an accordian is implemented as a child of its header?"

If you do this, then the entire accordion panel, including all links and sub-accordions if present, plus all active elements and form fields and headings and structural markup, are all stuffed into the header, making it impossible to keep track of an intuitive document order, because it would be like trying to nest multiple heading elements inside of each other. E.G Like an H3 inside of an H2 inside of an H1, etc.

**Comment by electronicwoft** [comment]
heheh .. yeah, sorry - got ahead of myself there ... of course that would be a mess ... I guess I am just questioning the need for yet another ARIA attribute that will be subject to patchy implementation and support for a widget whose related parts are adjacent.
